packages/fx-core/src/common/tools.ts

```diff
@@ -561,3 +561,79 @@ export function isSPFxProject(projectSettings?: ProjectSettings): boolean {
 export function getHashedEnv(envName: string): string {
   return crypto.createHash("sha256").update(envName).digest("hex");
 }
+
+interface BasicJsonSchema {
+  type: string;
+  properties?: {
+    [k: string]: unknown;
+  };
+}
+function isBasicJsonSchema(jsonSchema: unknown): jsonSchema is BasicJsonSchema {
+  if (!jsonSchema || typeof jsonSchema !== "object") {
+    return false;
+  }
+  return typeof (jsonSchema as { type: unknown })["type"] === "string";
+}
+
+// Redact user content of "obj";
+//
+// DFS "obj" and "jsonSchema" together to redact the following things:
+// - properties that is not defined in jsonSchema
+// - the value of properties that is defined in jsonSchema, but the keys will remain
+// Example:
+//  Input:
+//  obj = {"name": "some name", "user defined property": {"key1": "value1"}}
+//  jsonSchema = {
+//    "type": "object",
+//    "properties": {
+//      "name": { "type": "string" }
+//    }
+//  }
+//  Output:
+//  {"name": null}
+export function _redactObject(
```

Comment:
This function seems to be internal, do we need to expose it?

Reply:
No. I forgot to delete the export.